File: fsharp/etl/Etl.fs
module Etl

    let transform (scoresWithLetters: Map<int, char list>): Map<char, int> =
        scoresWithLetters
        |> Map.toList
        |> List.collect (fun (score: int, letters: char list) -> letters |> List.map (fun (letter: char) -> (System.Char.ToLower(letter), score)))
        |> Map.ofList

File: solutions/scala/etl/1/src/main/scala/Etl.scala
object Etl
{
  private def extractLoad(score: Int, letters: Seq[String]) =
  {
    letters.map[(String, Int)]((letter: String) => (letter.toLowerCase, score))
  }

  def transform(scoreMap: Map[Int, Seq[String]]): Map[String, Int] =
  {
    scoreMap.flatMap[String, Int](Function.tupled(extractLoad))
  }
}
